api/types/constants.go

```diff
@@ -985,6 +985,24 @@ const (
 	PluginGenerationLabel = TeleportInternalLabelPrefix + "plugin-generation"
 )
 
+const (
+	// NotificationPageTokensDelimiter is the delimiter which separates the user-specific notifications page token from
+	// the global notifications page token in the PageTokens field of a request.
+	NotificationPageTokensDelimiter = ","
+	// NotificationsEndSignal is the next page token that is returned if the end of a notifications list has been reached.
+	// In other paginated APIs, this is usually just "", after which the caller knows to stop fetching. However, since we fetch from
+	// two lists at once, it's possible that we reach the end of one list but not the other, and thus the caller will keep fetching. In
+	// this scenario, using "" again as the page token for that list would start from its beginning again, and this cycle would repeat infinitely.
+	NotificationsEndSignal = "end"
```

Comment:
The end sentinel value is required so that the requester (ie. frontend) can tell the api to no longer list items from that particular list. What we typically do in other paginated api's is the frontend stops fetching as soon as the response returns `""` as the `nextKey`, however this would cause issues because `""` provided as a `startKey` is interpreted as the default `startKey` which starts from the beginning of the list.

Let's say a user has reached the end of the global notifications list, but still has more user-specific notifications, the `nextKey` in the response would be something like `user-notif-5,""`, and those are the `startKey`s which will be used in the subsequent request. However the `""` in that request will then be interpreted as the `startKey` for global notifications, and it'll start iterating from the beginning of the global notifications list again. Using an end sentinel value was the most straightforward way I could come up with to resolve this issue